src/dev/x11/lv_x11.h

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file lv_x11.h
+ *
+ */
+
+#ifndef LV_X11_H
+#define LV_X11_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "../../display/lv_display.h"
+#include "../../indev/lv_indev.h"
+
+#if LV_USE_X11
+
+/*********************
+ *      DEFINES
+ *********************/
+
+#ifndef LV_X11_DOUBLE_BUFFER
+#  define LV_X11_DOUBLE_BUFFER 1
+#endif
+#ifndef LV_X11_RENDER_MODE
+#  define LV_X11_RENDER_MODE   LV_DISPLAY_RENDER_MODE_PARTIAL
+#endif
+
+/** Header of private display driver user data - for internal use only */
+typedef struct {
+    struct _XDisplay   *  display;  /**< X11 display object     */
+    struct _x11_inp_data * inp_data; /**< input user data object */
+} _x11_user_hdr_t;
+
+/** optional window close callback function type
+ *  @see lv_x11_window_set_close_cb
+*/
+typedef void(*lv_x11_close_cb)(void * user_data);
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/**
+ * create and add keyboard, mouse and scrillwheel objects and connect them to x11 display.
+ *
+ * This is a convenience method handling the typical input initialisation of an X11 window:
+ * - create keyboard (@ref lv_x11_keyboard_create)
+ * - create mouse (with scrollwheel, @ref lv_x11_mouse_create @ref lv_x11_mousewheel_create)
+ *
+ * @param[in] disp      the created X11 display object from @ref lv_x11_window_create
+ * @param[in] mouse_img optional image description for the mouse cursor (NULL for no/invisible mouse cursor)
+ */
+void lv_x11_inputs_create(lv_display_t * disp, lv_image_dsc_t const * mouse_img);
+
+/**
+ * create the X11 display
+ *
+ * The minimal initialisation for initializing the X11 display driver with keyboard/mouse support:
+ * @code
+ * lv_display_t* disp = lv_x11_window_create("My Window Title", window_width, window_width);
+ * lv_x11_inputs_create(disp, NULL);
+ * @endcode
+ * or with mouse cursor icon:
+ * @code
+ * lv_img_dsc_t mouse_symbol = {.....};
+ * lv_display_t* disp = lv_x11_window_create("My Window Title", window_width, window_width);
+ * lv_x11_inputs_create(disp, &mouse_symbol);
+ * @endcode
+ *
+ * @param[in] title    title of the created X11 window
+ * @param[in] hor_res  horizontal resolution (=width) of the X11 window
+ * @param[in] ver_res  vertical resolution (=height) of the X11 window
+ * @return             pointer to the display object
+ */
+lv_display_t * lv_x11_window_create(char const * title, int32_t hor_res, int32_t ver_res);
+
+/**
+ * set optional application callback to get informed on X11 window close event (to cleanup application)
+ * @param[in] disp      the created X11 display object from @lv_x11_window_create
+ * @param[in] close_cb  callback funtion to be called
+ */
+void lv_x11_window_set_close_cb(lv_display_t * disp, lv_x11_close_cb close_cb, void * user_data);
```

Comment:
Can we use an `LV_EVENT_DELETED` event instead of a dedicated callback?

Reply:
LV_EVENT_DELETED as I've understood is to delete objects (e.g. widgets?).
lv_x11_window_set_close_cb() callback is triggered from the window manager close event. The idea is to give the application the possibility to close the window (and its objects) properly (instead of "terminated" by the display driver). In the example refernced I leave the main loop after this [event](https://github.com/gcopoix/lv_port_pc_vscode/blob/feature/v9/main/src/main.c#L53) and call [lv_display_remove()](https://github.com/gcopoix/lv_port_pc_vscode/blob/feature/v9/main/src/main.c#L129) to properly shutdown the display object, so the application can exit properly.
What do you think?

Reply:
This is how I handled it in the SDL driver:
1. I registered an [`LV_EVENT_DELETE` callback](https://github.com/lvgl/lvgl/blob/master/src/dev/sdl/lv_sdl_window.c#L89) on the display for clean up. So when `lv_display_remove` is called either by the user or when the window is closed the display will clean up properly.  
2. Remove the display in the Window close event. See [here](https://github.com/lvgl/lvgl/blob/master/src/dev/sdl/lv_sdl_window.c#L227-L228).
 
 I think the same can be applied for X11 too. 
 
 If you would like to terminate the main loop as well, you can register an callback for the display when you have created, and set a global variable in it to tell main loop to stop.
 
 What do you think?
 
 > I haven't found any usage of LV_EVENT_DELETED, did you mean LV_EVENT_DELETE (without 'D')?
 
 Sorry, yes, I  meant `LV_EVENT_DELETE`.

Reply:
1. rebased on current master
2. added initial documentation in docs/integration/driver/display tree
3. replaced own close_cb with LV_EVENT_DELETE callback

What's wrong with the pipeline (Micropython CI / Build unix port (pull_request)) ?

Reply:
> What's wrong with the pipeline (Micropython CI / Build unix port (pull_request)) ?

It's unrelated, I hope it will be resolved soon.